app/containers/write.js

```diff
@@ -93,7 +94,38 @@ class writeScreen extends Component {
   componentDidUpdate (prevProps) {
     if (this.props.currentChannel !== prevProps.currentChannel) {
       this.focusInput()
-    }
+      var i = 0 
+      var flag = 0
+      for(i = 0; i < this.myArray.length; i++){
+        if (prevProps.currentChannel == this.myArray[i].name){
+          if (this.textInput.value != this.myArray[i].text){
+            window[this.props.currentChannel] = {
+              name: prevProps.currentChannel,
+              text: this.textInput.value,
+            }
+            this.myArray.splice(i,1)
+            this.myArray.push(window[this.props.currentChannel])
+          }else{
+            flag = 1
```

Comment:
Can you use more descriptive variable names? myArray and flag may get confusing for others if their names are not descriptive of what they do.